ios/engine/KMEI/KeymanEngine/Classes/KeymanPackage.swift

```diff
@@ -124,22 +124,22 @@ public class KeymanPackage {
     
     return nil
   }
-  
+
+  @available(*, deprecated, message: "Use of the completion block is unnecessary; this method now returns synchronously.")
   static public func extract(fileUrl: URL, destination: URL, complete: @escaping (KeymanPackage?) -> Void) throws {
     try unzipFile(fileUrl: fileUrl, destination: destination) {
       complete(KeymanPackage.parse(destination))
     }
   }
 
-  static public func unzipFile(fileUrl: URL, destination: URL, complete: @escaping () -> Void) throws {
-    try Zip.unzipFile(fileUrl, destination: destination, overwrite: true,
-                      password: nil,
-                      progress: { (progress) -> () in
-                        //TODO: add timeout
-                        if(progress == 1.0) {
-                          complete()
-                        }
-                      })
```

Comment:
Yeah, so this bit right here was the heart of the issue.  The... original implementer... believed that this was an asynchronous method.  It turns out that this is untrue; investigating the documentation for our [Zip dependency](https://github.com/marmelroy/Zip#advanced-zip) shows that it's a _progress_ handler.  The function itself is synchronous.

Realizing that one detail cascaded into the ability to remove most all of our Package-related completion handler parameters.